### PR TITLE
Create ParticipantsSearch component and use it in Participants form

### PR DESCRIPTION
**Pull request by Marce1ina**
#### Story / Bug id:
https://github.com/CodeForPoznan/codeforpoznan.pl_v3/issues/56

#### Description:
Added ParticipantsSearch component and used it in Participants form to replace v-select.

#### Migrations:
N/A

#### New imports / dependencies:
N/A

#### What tests do I need to run to validate this change:

1. Go to 'Edit existing participant' form and make sure that participant can be found in searchbox by any related data (slack, github, name, last name, email, phone number, id). 
2. Sanity testing of all the functionalities in this form (ie. you can edit participant, save changes, then search by changed data; you can clear the form; you can switch between 'edit' and 'add' mode without any disturbance).

### CODE REVIEW

**Review by KSodowska — APPROVED** [review]
@Marce1ina lookin' good :+1:

File: frontend/src/components/ParticipantsSearch.vue
Comment (OtisRed) [inline frontend/src/components/ParticipantsSearch.vue:6]: I know that you replicated the previous solution here but it would make user's life much easier if we could render/display results for each item in the format:
"`first_name` `last_name` (`slack`/`github`)"

In the documentation I found that combobox accepts stings:
https://vuetifyjs.com/en/api/v-combobox/#props
Not sure if that applies to the example above. 

Yet, given that we will use this component for different purposes and there are three different sources of information about who selectedParticipant is it would be great to have all that information on the list not to wonder about if the person with slack nick A is also the person with github nick B (and they do vary)

File: frontend/src/components/ParticipantsSearch.vue
Comment (OtisRed) [inline frontend/src/components/ParticipantsSearch.vue:28]: If the second would be feasible we could use `first_name` to sort participants instead of github. 
(Knowing someone by name comes naturally and if the pattern I described before is possible, then this additional information would make it easy to distinguish between different people having the same name)

**Review by OtisRed — CHANGES_REQUESTED** [review]
One technical comments and two regarding user experience :)

**Comment by Marce1ina** [comment]
@OtisRed thanks for your input, I've addressed the comments, please check if you're fine with the new behaviour.

**Comment by OtisRed** [comment]
@Marce1ina thanks! It's much easier to navigate now! However two more points before we merge it:
1. I guess it's important to figure out the consistent solution between this and #384 so if possible please talk about it with @stanislawK.
2. Would it be possible to show "no results" in case when the input has no match? Having information that there is no information can be also important information :D Also I think this might be related to the discussion you have with @stanislawK about filtering and sorting functions.

**Comment by Marce1ina** [comment]
@OtisRed added 'no results' message. The consistency issue will be addressed separately, as discussed on priv.
Will need your approval to merge :)

**Review by OtisRed — APPROVED** [review]
Perfect!

**Pull request merged**
